### DISCUSSION TITLE
Get about 15 minutes into a stream and game freezes on client, still accessible on host, attempts to reconnect to client produce errors

### DISCUSSION THREAD

**Discussion by Travillion** [LizardByte/community#396] [Sunshine]
My host PC is Win 11, intel i9-14900 with NVIDIA 4070 ti Super. I have GeForce Experience uninstalled. Client is NVIDIA Shield TV, hooked up to my 4k TV (this all on wired LAN). Since my host monitor is only 1080p, I have a virtual display that is turned off by default. In my Sunshine configuration, I use DisplaySwitch Do and Undo commands to choose the correct monitor on start/close.  I don't have any issues with the monitor selection beginning or exiting a stream. However, I cannot make it more than ~15 minutes into a stream without the client screen freezing (audio is usually still active), and the game is active on the host monitor. If I look at the sunshine web UI, there are no sunshine errors thrown that I can see. Exiting and re-entering moonlight produces errors in sunshine, usually along the lines of no encoder can be found. I usually end up finishing my gaming session on my host pc because no amount of resets and force closes (sunshine and moonlight) can really get me back into my stream for more than a few seconds. I don't know if this is a sunshine issue, moonlight issue, or virtual display issue. I can provide any logs that would be helpful, just let me know which log level to provide. 
Here is my config file:

origin_web_ui_allowed = pc
min_log_level = 2
av1_mode = 1
gamepad = x360
address_family = both
nvenc_preset = 4
qp = 25
global_prep_cmd = [{"do":"DisplaySwitch 4","undo":"DisplaySwitch 1","elevated":false}]
nvenc_realtime_hags = enabled

**Comment by Travillion** [comment]
Upon further research, it appears this is the good old HAGS, high VRAM usage bug. Disabling HAGS eliminated the freeze, although increased stuttering quite a bit unless I lower the resolution. Interestingly enough, I didn't experience this the first time I ran Jedi: Survivor, which was the night before the latest Nvidia studio driver update. I know this bug is tightly linked with Nvidia driver issues. I am rolling back my driver to see if I can enable HAGS and play reliably.

**Reply by Travillion** [reply]
Whelp, I made it longer (~45 minutes?) on the old driver (551.23), but in the end I still got the freeze. Looks like I need to continue experimenting with lowering my settings and/or disabling HAGS. Frustrating because the game seems to run really well until it freezes out of the blue. Wish I could keep the current settings but streaming is way too choppy without HAGS.

**Comment by Dijkie85** [comment]
Hey, did you find a reliable solution? I just switched from GS to Sunshine and facing a similar issue as I had in the past, but was under the impression this HAGS issue had been fixed in the latest releases...

**Comment by Travillion** [comment]
HAGS issue was fixed as much as sunshine team can because there are
underlying Nvidia driver issues. Myself and others still have this issue if
HAGS is enabled. I've never had an issue with HAGS disabled.

**Comment by QuentinSc** [comment]
This bug is back :(

[a comment by an automated account is not shown]
